initial push for hierarcy based permissions

natlibfi-jonollil opened the pull request.

natlibfi-arlehiko commented on src/interfaces/utils.js:
```diff
@@ -25,14 +25,33 @@
-	return permitted;
-}
+export function hasPermission(type, command, profile, bgroups = []) {
+	const permintted = profile.groups.some(
```
Fix typo: `permintted` -> `permitted`

natlibfi-arlehiko commented on src/interfaces/utils.js:
```diff
@@ -25,14 +25,33 @@
-	});
-	return permitted;
-}
+export function hasPermission(type, command, profile, bgroups = []) {
```
Change parameter names to be more clear:

- *profile* -> **userGroups**
- *bgroups* -> **permittedGroups**

Also, it's unecessary to pass the whole profile object here. Passing `profile.auth.groups` is sufficient.

natlibfi-arlehiko commented on src/interfaces/utils.js:
```diff
@@ -25,14 +25,33 @@
 
-export function hasAdminPermission(user) {
-	return hasPermission({auth: {groups: ['admin']}}, user);
+	return profile.groups.some(group => bgroups.includes(group)) && permintted;
```
Optimization: Turn this the other way around nt reduce unnecessary array iteration if the boolean doesn't match:
`return permitted  && profile.groups.some(group => bgroups.includes(group))`

natlibfi-jonollil commented:
> Doesn't seem to check permission to use the `abort` operation of updateBlob?

It needed `UPDATE` tier permissions to `ABORT`blob process. Fixed, updated tests and tested.

natlibfi-arlehiko commented on src/interfaces/utils.js:
```diff
@@ -45,13 +45,24 @@ const permissions = {
 };
 
 export function hasPermission(type, command, userGroups, permittedGroups = []) {
```
Make this more easier to read:
```js
const commandPermissions = permissions[type][command];

if (userGroups.includes('system')) {
  return true;

if (hasGroup(permittedGroups) === false) {
  return false;
}

if (commandPermissions.includes('all') || hasGroup(commandPermissions)) {
  return true;
}

function hasGroup(permitted) {
  return userGroups.some(g => permitted.includes(g));
}
```

natlibfi-jonollil commented:
https://github.com/NatLibFi/melinda-record-import-api/pull/14 merge first if possible?
